features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Definitions/AsyncApi/AsyncApiDefinitionUI.jsx

```diff
@@ -0,0 +1,177 @@
+    },
+    listRoot: {
+        paddingTop: 0,
+    },
+    nested: {
+        paddingLeft: theme.spacing(3),
+        paddingTop: 3,
+        paddingBottom: 3,
+    },
+    childList: {
+        paddingTop: 0,
+        marginTop: 0,
+        paddingBottom: 0,
+    },
+    contentWrapper: {
+        maxWidth: theme.custom.contentAreaWidth,
+        paddingLeft: theme.spacing(3),
+        paddingTop: theme.spacing(3),
+        paddingRight: theme.spacing(3),
+    },
+    titleSub: {
+        marginLeft: theme.spacing(3),
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+        color: theme.palette.getContrastText(theme.palette.background.default),
+    },
+    generateCredentialWrapper: {
+        marginLeft: 0,
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+    },
+    genericMessageWrapper: {
+        margin: theme.spacing(2),
+    },
+    typeText: {
+        color: '#000',
+    },
+    docLinkRoot: {
+        paddingLeft: 0,
+    },
+    toggler: {
+        height: '100%',
+        paddingTop: 20,
+        cursor: 'pointer',
+        marginLeft: '-20px',
+        display: 'block',
+    },
+    togglerTextParent: {
+        writingMode: 'vertical-rl',
+        transform: 'rotate(180deg)',
+    },
+    togglerText: {
+        textOrientation: 'sideways',
+    },
+    toggleWrapper: {
+        position: 'relative',
+        background: '#fff9',
+        paddingLeft: 20,
+    },
+    docsWrapper: {
+        margin: 0,
+    },
+    docContainer: {
+        display: 'flex',
+        marginLeft: theme.spacing(3),
+        marginRight: theme.spacing(2),
+        marginTop: theme.spacing(2),
+    },
+    docListWrapper: {
+        width: 285,
+    },
+    docView: {
+        flex: 1,
+    },
+    listItemRoot: {
+        minWidth: 30,
+    },
+    editorPane: {
+        width: '100%',
+        maxHeight: '100vh',
+        minHeight: '100vh',
+        overflow: 'auto',
+    },
+    editorRoot: {
+        height: '100%',
+    },
+});
+
+
+class AsyncApiDefinitionUI extends Component {
+    static contextType = ApiContext;
+
+    constructor(props) {
+        super(props);
+        this.state = { spec: '' };
+    }
+
+    componentDidMount() {
+        const context = this.context;
+        this.setState({ spec: context.api.apiDefinition });
+    }
+
+    render() {
+        const { spec } = this.state;
+        const { classes } = this.props;
+        return (
+            <>
+                <Typography variant='h4' className={classes.titleSub}>
+                    <FormattedMessage
+                        id='Apis.Details.Async.Definition.title'
+                        defaultMessage='AsyncAPI Specification'
+                    />
+                </Typography>
+                <Grid container spacing={1} className={classes.editorRoot}>
+                    <Grid item className={classes.editorPane}>
+                        <AsyncApiComponent schema={spec} />
```

Comment:
No need to get the spec from the context and pass. You can simply get the same within the AsyncAPIComponent.